pkg/server/api/entry/v1/service_test.go

```diff
@@ -799,6 +799,448 @@ func setupServiceTest(t *testing.T, ds datastore.DataStore) *serviceTest {
+					},
+				},
+			},
+		},
+		{
+			name:           "Success Don't Update TTL",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask:      &types.EntryMask{
+				// With this empty, the update operation should be a no-op
+			},
+			outputMask: &types.EntryMask{
+				Ttl: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					Ttl: 500000,
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						Ttl: 60,
+					},
+				},
+			},
+		},
+		{
+			name:           "Fail Invalid Spiffe Id",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				SpiffeId: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					SpiffeId: &types.SPIFFEID{TrustDomain: "", Path: "/invalid"},
+				},
+			},
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "failed to convert entry",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "invalid spiffe ID: spiffeid: trust domain is empty",
+						"spiffe_id":     "path:\"/invalid\" ",
+					},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.InvalidArgument),
+						Message: "failed to convert entry: invalid spiffe ID: spiffeid: trust domain is empty"},
+				},
+			},
+		},
+		{
+			name:           "Fail Invalid Parent Id",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				ParentId: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					ParentId: &types.SPIFFEID{TrustDomain: "", Path: "/invalid"},
+				},
+			},
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "failed to convert entry",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "invalid parent ID: spiffeid: trust domain is empty",
+					},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.InvalidArgument),
+						Message: "failed to convert entry: invalid parent ID: spiffeid: trust domain is empty"},
+				},
+			},
+		},
+		{
+			name:           "Fail Datastore Error",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				ParentId: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					ParentId: &types.SPIFFEID{TrustDomain: "", Path: "/invalid"},
+				},
+			},
+			dsError: errors.New("some error"),
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "failed to fetch entry",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "some error",
+					},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.Internal), Message: "failed to fetch entry"},
+				},
+			},
+		},
+		{
+			name:           "Check Output Masks Work",
```

Comment:
can you add a cases for:
- nil iinput mask
- nil output mask
- all false  input mask
- all false output mask?

Reply:
Good idea! Done